client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -303,23 +326,15 @@ public void flush() {
     
     private void flushInternal() {
         boolean success = false;
-        String sealedSegment = null;
         while (!success) {
             success = true;
```

Comment:
Success is not used and the semantic is currently lost.
The contract of flush is that there are no unwritten events when the call returns. However any write may encounter a segment sealed. So the segmentSealedCallback must be able to run in parallel to flush, but if it does and the data is re-sent to a new segment, it may not be yet flushed, in which case flush failed to achieve its goal. 
We could perhaps add an isSealed method to SegmentOutputStream to help with this. But even that is not so simple, as we need a way to wait for any resending resulting from a segment seal.